### PR TITLE
[Core][Spill manager refactor 0] Store ref-count in a separate store

### PR DESCRIPTION
**Pull request by scv119**
This store reference counter in LifecycleMetadataStore, other than object store. This opens up the door for follow up PRs where we store spilled/primary info as well.

Test plan
- [x] existing tests

### CODE REVIEW

File: src/ray/object_manager/plasma/lifecycle_meta_store.cc [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+// under the License.
+
+#include "ray/object_manager/plasma/lifecycle_meta_store.h"
+
+namespace plasma {
+
+using namespace ray;
+
+namespace {
+const LifecycleMetadata kDefaultStats{};
```
Comment (MissiontoMars) [inline src/ray/object_manager/plasma/lifecycle_meta_store.cc:25]: Do we need to judge whether a `LifecycleMetadata` object is the default `kDefaultStats`?
`const auto &meta = lifecycleMetadataStore.GetMetadata();
if (meta == kDefaultStats) {
}`

File: src/ray/object_manager/plasma/lifecycle_meta_store.cc [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+bool LifecycleMetadataStore::IncreaseRefCount(const ObjectID &id) {
```
Comment (MissiontoMars) [inline src/ray/object_manager/plasma/lifecycle_meta_store.cc:28]: Return value seems useless.

File: src/ray/object_manager/plasma/object_lifecycle_manager.h [hunk 3]
```diff
@@ -92,6 +93,9 @@ class IObjectLifecycleManager {
   ///
   /// \return true if object exists and reference count is greater than 0, false otherise.
   virtual bool RemoveReference(const ObjectID &object_id) = 0;
+
+  /// Returns true if object is spillable.
+  virtual bool IsObjectSpillable(const ObjectID &object_id) const = 0;
```
Comment (MissiontoMars) [inline src/ray/object_manager/plasma/object_lifecycle_manager.h:98]: Could you add more comments about `spillable`? An object which has been evicted and ref count is 1

**Comment by scv119** [comment]
This seems to be overkill...

**Pull request closed without merging**